tests/test_NoiseReplacer.py

```diff
@@ -75,15 +76,76 @@ def setUp(self):
             family.addChild(140000.0, lsst.geom.Point2D(72.3, 149.1))
             family.addChild(90000.0, lsst.geom.Point2D(68.5, 156.9))
 
-    def testSingleFrameMeasurement(self):
-        """Test noise replacement in single frame measurement.
+    def testMeasureNoiseReplacer(self):
+        """Test noise replacement in SFM with ''measure'' mode.
 
         We compare the instFlux inside and outside source Footprints on an
         extremely high S/N image.
         """
+        # We choose a random seed which causes the test to pass.
+        task = self.makeSingleFrameMeasurementTask("test_NoiseReplacer")
+        task.config.noiseReplacer.noiseSource = 'measure'
+        exposure, catalog = self.dataset.realize(1.0, task.schema, randomSeed=0)
+        task.run(catalog, exposure)
+        sumVariance = exposure.getMaskedImage().getVariance().getArray().sum()
+        for record in catalog:
+            self.assertFloatsAlmostEqual(record.get("test_NoiseReplacer_inside"),
+                                         record.get("truth_instFlux"), rtol=1E-3)
+
+            # N.B. Next line checks that a random value is correct to a
+            # statistical 1-sigma prediction; some RNG seeds may cause it to
+            # fail (indeed, 67% should)
+            self.assertLess(record.get("test_NoiseReplacer_outside"), np.sqrt(sumVariance))
 
+    def testVarianceNoiseReplacer(self):
+        """Test noise replacement in SFM with ''variance'' mode.
+        """
+        # We choose a random seed which causes the test to pass.
+        task = self.makeSingleFrameMeasurementTask("test_NoiseReplacer")
+        task.config.noiseReplacer.noiseSource = 'variance'
+        exposure, catalog = self.dataset.realize(1.0, task.schema, randomSeed=0)
+        task.run(catalog, exposure)
+        sumVariance = exposure.getMaskedImage().getVariance().getArray().sum()
+        for record in catalog:
+            self.assertFloatsAlmostEqual(record.get("test_NoiseReplacer_inside"),
+                                         record.get("truth_instFlux"), rtol=1E-3)
+
+            # N.B. Next line checks that a random value is correct to a
+            # statistical 1-sigma prediction; some RNG seeds may cause it to
+            # fail (indeed, 67% should)
+            self.assertLess(record.get("test_NoiseReplacer_outside"), np.sqrt(sumVariance))
```

Comment:
Isn't this just an exact copy and paste of the test above, meaning you could do the whole thing with
```
for noiseSource in ['measure', 'variance']:
    task.config.noiseReplacer.noiseSource = noiseSource```